### ISSUE TITLE
Can't install `transgrp` package in some circumstances

### ISSUE THREAD

**Issue by james-d-mitchell** [gap-packages/PackageManager#80]
This came up in the docker container:

https://github.com/james-d-mitchell/gap-docker-minimal-arm

when trying to install the `transgrp` package using:

~~~
InstallPackage(\"transgrp\");
~~~

here:

https://github.com/james-d-mitchell/gap-docker-minimal-arm/blob/28d0ab04329a9c4ea8a005b4b3f1d64aad664029/Dockerfile#L38

This fails with:

~~~
gap@85ffaa889f82:~$ gap --bare
 *********   GAP 4.11.1 of 2021-03-02
 *  GAP  *   https://www.gap-system.org
 *********   Architecture: aarch64-unknown-linux-gnu-default64-kv7
 Configuration:  gmp 6.2.0, GASMAN, readline
 Loading the library and packages ...
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
gap> LoadPackage("packageman");
-----------------------------------------------------------------------------------------------------------------------------------
Loading  GAPDoc 1.6.4 (a meta package for GAP documentation)
by Frank L�beck (http://www.math.rwth-aachen.de/~Frank.Luebeck) and
   Max Neunh�ffer (http://www-groups.mcs.st-and.ac.uk/~neunhoef/).
maintained by:
   Frank L�beck (http://www.math.rwth-aachen.de/~Frank.Luebeck).
Homepage: http://www.math.rwth-aachen.de/~Frank.Luebeck/GAPDoc
Report issues at https://github.com/frankluebeck/GAPDoc/issues
-----------------------------------------------------------------------------------------------------------------------------------
-----------------------------------------------------------------------------------------------------------------------------------
Loading  PackageManager 1.2 (Easily download and install GAP packages)
by Michael Young (https://mct25.host.cs.st-andrews.ac.uk/).
Homepage: https://gap-packages.github.io/PackageManager/
Report issues at https://github.com/gap-packages/PackageManager/issues
-----------------------------------------------------------------------------------------------------------------------------------
true
gap> InstallPackage("transgrp");
#I  Getting PackageInfo URLs...
#I  Command wget not found
#I  Retrieving PackageInfo.g from https://www.math.colostate.edu/~hulpke/transgrp/PackageInfo.g ...
#I  Command wget not found
#I  Downloading archive from URL https://www.math.colostate.edu/~hulpke/transgrp/transgrp3.5.tar.gz ...
#I  Command wget not found
#I  Saved archive to /tmp/tmcw4I4p/transgrp3.5.tar.gz.pkgman
#I  Archive should contain 1 directory (not 2)
false
gap>
~~~

Which is triggered by the following line in the package manager:

https://github.com/gap-packages/PackageManager/blob/fd80d4668ca3685fdf0ee401d6edf91f1426ece1/gap/PackageManager.gi#L266

The output of:

https://github.com/gap-packages/PackageManager/blob/fd80d4668ca3685fdf0ee401d6edf91f1426ece1/gap/PackageManager.gi#L265

in this case is:

~~~
["", "transgrp"]
~~~

because `files` is:

~~~
[ "transgrp/LICENSE", "transgrp/PackageInfo.g", "transgrp/data/", "transgrp/data/trans40b.grp", "transgrp/data/trans36d.grp",
  "transgrp/data/trans42j.grp", "transgrp/data/trans40u.grp", "transgrp/data/trans36s.grp", "transgrp/data/trans16f.grp",
  "transgrp/data/trans40aj.grp", "transgrp/data/trans40bf.grp", "transgrp/data/trans20d.grp", "transgrp/data/trans25.grp",
  "transgrp/data/trans31.grp", "transgrp/data/trans19.grp", "transgrp/data/trans18.grp", "transgrp/data/trans30.grp",
  "transgrp/data/trans21a.grp", "transgrp/data/trans24.grp", "transgrp/data/trans40bg.grp", "transgrp/data/trans40ak.grp",
  "transgrp/data/trans16g.grp", "transgrp/data/trans40t.grp", "transgrp/data/trans36r.grp", "transgrp/data/trans42k.grp",
  "transgrp/data/trans40c.grp", "transgrp/data/trans36e.grp", "transgrp/data/trans37a.grp", "transgrp/data/trans36g.grp",
  "transgrp/data/trans40a.grp", "transgrp/data/trans36p.grp", "transgrp/data/trans40v.grp", "transgrp/data/trans42i.grp",
  "transgrp/data/trans16e.grp", "transgrp/data/trans40ai.grp", "transgrp/data/trans40be.grp", "transgrp/data/trans26.grp",
  "transgrp/data/trans27.grp", "transgrp/data/trans33.grp", "transgrp/data/trans40bd.grp", "transgrp/data/trans40ah.grp",
  "transgrp/data/trans16d.grp", "transgrp/data/trans42h.grp", "transgrp/data/trans36q.grp", "transgrp/data/trans40w.grp",
  "transgrp/data/trans36f.grp", "transgrp/data/trans40s.grp", "transgrp/data/trans36u.grp", "transgrp/data/trans42l.grp",
  "transgrp/data/trans40d.grp", "transgrp/data/trans36b.grp", "transgrp/data/trans40al.grp", "transgrp/data/trans37.grp",
  "transgrp/data/trans20b.grp", "transgrp/data/trans23.grp", "transgrp/data/trans22.grp", "transgrp/data/trans20c.grp",
  "transgrp/data/trans36.grp", "transgrp/data/trans40am.grp", "transgrp/data/trans40az.grp", "transgrp/data/trans40ba.grp",
  "transgrp/data/trans16a.grp", "transgrp/data/trans42z.grp", "transgrp/data/trans41a.grp", "transgrp/data/trans40e.grp",
  "transgrp/data/trans36c.grp", "transgrp/data/trans42m.grp", "transgrp/data/trans40r.grp", "transgrp/data/trans36t.grp",
  "transgrp/data/trans42o.grp", "transgrp/data/trans36v.grp", "transgrp/data/trans40p.grp", "transgrp/data/trans42x.grp",
  "transgrp/data/trans36a.grp", "transgrp/data/trans40g.grp", "transgrp/data/trans16c.grp", "transgrp/data/trans40bc.grp",
  "transgrp/data/trans40ax.grp", "transgrp/data/trans40ao.grp", "transgrp/data/trans20a.grp", "transgrp/data/trans20.grp",
  "transgrp/data/trans34.grp", "transgrp/data/trans35.grp", "transgrp/data/trans21.grp", "transgrp/data/trans40an.grp",
  "transgrp/data/trans40ay.grp", "transgrp/data/trans40bb.grp", "transgrp/data/trans16b.grp", "transgrp/data/trans40f.grp",
  "transgrp/data/trans42y.grp", "transgrp/data/trans36w.grp", "transgrp/data/trans40q.grp", "transgrp/data/trans42n.grp",
  "transgrp/data/trans30g.grp", "transgrp/data/trans46a.grp", "transgrp/data/trans45z.grp", "transgrp/data/trans30p.grp",
  "transgrp/data/trans45m.grp", "transgrp/data/trans24bw.grp", "transgrp/data/trans24al.grp", "transgrp/data/trans24cd.grp",
  "transgrp/data/trans46.grp", "transgrp/data/trans27c.grp", "transgrp/data/trans24x.grp", "transgrp/data/trans24o.grp",
  "transgrp/data/trans24n.grp", "transgrp/data/trans24y.grp", "transgrp/data/trans27b.grp", "transgrp/data/trans47.grp",
  "transgrp/data/trans24ba.grp", "transgrp/data/trans24az.grp", "transgrp/data/trans24ce.grp", "transgrp/data/trans24am.grp",
  "transgrp/data/trans24bv.grp", "transgrp/data/trans30q.grp", "transgrp/data/trans45l.grp", "transgrp/data/trans44h.grp",
  "transgrp/data/trans30f.grp", "transgrp/data/trans45y.grp", "transgrp/data/trans30d.grp", "transgrp/data/trans45n.grp",
  "transgrp/data/trans30s.grp", "transgrp/data/trans24bt.grp", "transgrp/data/trans24ao.grp", "transgrp/data/trans24ax.grp",
  "transgrp/data/trans24bc.grp", "transgrp/data/trans45.grp", "transgrp/data/trans24l.grp", "transgrp/data/trans24m.grp",
  "transgrp/data/trans44.grp", "transgrp/data/trans24z.grp", "transgrp/data/trans27a.grp", "transgrp/data/trans24cf.grp",
  "transgrp/data/trans24bb.grp", "transgrp/data/trans24ay.grp", "transgrp/data/trans24an.grp", "transgrp/data/trans24bu.grp",
  "transgrp/data/trans45o.grp", "transgrp/data/trans30r.grp", "transgrp/data/trans31a.grp", "transgrp/data/trans45x.grp",
  "transgrp/data/trans30e.grp", "transgrp/data/trans45k.grp", "transgrp/data/trans30a.grp", "transgrp/data/trans45ak.grp",
  "transgrp/data/trans24bf.grp", "transgrp/data/trans24cb.grp", "transgrp/data/trans24bq.grp", "transgrp/data/trans24aj.grp",
  "transgrp/data/trans24i.grp", "transgrp/data/trans27e.grp", "transgrp/data/transminimals.grp", "transgrp/data/trans26a.grp",
  "transgrp/data/trans40.grp", "transgrp/data/trans41.grp", "transgrp/data/trans27d.grp", "transgrp/data/trans24h.grp",
  "transgrp/data/trans24ak.grp", "transgrp/data/trans24bp.grp", "transgrp/data/trans24cc.grp", "transgrp/data/trans24bg.grp",
  "transgrp/data/trans45aj.grp", "transgrp/data/trans45j.grp", "transgrp/data/trans45h.grp", "transgrp/data/trans30b.grp",
  "transgrp/data/trans45ah.grp", "transgrp/data/trans24ca.grp", "transgrp/data/trans24be.grp", "transgrp/data/trans24br.grp",
  "transgrp/data/trans24ai.grp", "transgrp/data/trans24j.grp", "transgrp/data/trans43.grp", "transgrp/data/trans27f.grp",
  "transgrp/data/trans27g.grp", "transgrp/data/trans42.grp", "transgrp/data/trans24k.grp", "transgrp/data/trans24ah.grp",
  "transgrp/data/trans24bs.grp", "transgrp/data/trans24bd.grp", "transgrp/data/trans45ai.grp", "transgrp/data/trans30c.grp",
  "transgrp/data/trans47a.grp", "transgrp/data/trans45i.grp", "transgrp/data/trans30t.grp", "transgrp/data/trans45d.grp",
  "transgrp/data/trans45s.grp", "transgrp/data/trans30n.grp", "transgrp/data/trans45ad.grp", "transgrp/data/trans24ar.grp",
  "transgrp/data/trans24bi.grp", "transgrp/data/trans28e.grp", "transgrp/data/trans24ae.grp", "transgrp/data/trans24f.grp",
  "transgrp/data/trans24q.grp", "transgrp/data/trans24p.grp", "transgrp/data/trans24g.grp", "transgrp/data/trans24ad.grp",
  "transgrp/data/trans24bh.grp", "transgrp/data/trans24as.grp", "transgrp/data/trans28d.grp", "transgrp/data/trans45ae.grp",
  "transgrp/data/trans45r.grp", "transgrp/data/trans30o.grp", "transgrp/data/trans44a.grp", "transgrp/data/trans45e.grp",
  "transgrp/data/trans44c.grp", "transgrp/data/trans33a.grp", "transgrp/data/trans45g.grp", "transgrp/data/trans30m.grp",
  "transgrp/data/trans45p.grp", "transgrp/data/trans45ag.grp", "transgrp/data/trans28f.grp", "transgrp/data/trans24aq.grp",
  "transgrp/data/trans24bj.grp", "transgrp/data/trans24af.grp", "transgrp/data/trans25a.grp", "transgrp/data/trans24e.grp",
  "transgrp/data/trans24r.grp", "transgrp/data/trans24s.grp", "transgrp/data/trans27h.grp", "transgrp/data/trans24d.grp",
  "transgrp/data/trans24ag.grp", "transgrp/data/trans28g.grp", "transgrp/data/trans24bk.grp", "transgrp/data/trans24ap.grp",
  "transgrp/data/trans45af.grp", "transgrp/data/trans30l.grp", "transgrp/data/trans45q.grp", "transgrp/data/trans45f.grp",
  "transgrp/data/trans44b.grp", "transgrp/data/trans45u.grp", "transgrp/data/trans30h.grp", "transgrp/data/trans44f.grp",
  "transgrp/data/trans45b.grp", "transgrp/data/trans8.grp", "transgrp/data/trans45ab.grp", "transgrp/data/trans24bx.grp",
  "transgrp/data/trans24ac.grp", "transgrp/data/trans24at.grp", "transgrp/data/trans24bo.grp", "transgrp/data/trans28c.grp",
  "transgrp/data/trans24w.grp", "transgrp/data/trans24a.grp", "transgrp/data/trans24v.grp", "transgrp/data/trans24bn.grp",
  "transgrp/data/trans24au.grp", "transgrp/data/trans28b.grp", "transgrp/data/trans24ab.grp", "transgrp/data/trans24by.grp",
  "transgrp/data/trans45ac.grp", "transgrp/data/trans9.grp", "transgrp/data/trans45c.grp", "transgrp/data/trans44g.grp",
  "transgrp/data/trans45t.grp", "transgrp/data/trans30i.grp", "transgrp/data/trans30k.grp", "transgrp/data/trans45v.grp",
  "transgrp/data/trans45a.grp", "transgrp/data/trans44e.grp", "transgrp/data/MIN.gz", "transgrp/data/trans45aa.grp",
  "transgrp/data/trans24aw.grp", "transgrp/data/trans24bl.grp", "transgrp/data/trans24t.grp", "transgrp/data/trans24c.grp",
  "transgrp/data/trans24b.grp", "transgrp/data/trans24u.grp", "transgrp/data/trans28a.grp", "transgrp/data/trans24bm.grp",
  "transgrp/data/trans24av.grp", "transgrp/data/trans24aa.grp", "transgrp/data/trans24bz.grp", "transgrp/data/trans44d.grp",
  "transgrp/data/trans30j.grp", "transgrp/data/trans45w.grp", "transgrp/data/trans39b.grp", "transgrp/data/trans35a.grp",
  "transgrp/data/trans42c.grp", "transgrp/data/trans36m.grp", "transgrp/data/trans40k.grp", "transgrp/data/trans42t.grp",
  "transgrp/data/trans18d.grp", "transgrp/data/trans42aa.grp", "transgrp/data/trans40at.grp", "transgrp/data/trans40ac.grp",
  "transgrp/data/trans10.grp", "transgrp/data/trans38.grp", "transgrp/data/trans39.grp", "transgrp/data/trans11.grp",
  "transgrp/data/trans40ab.grp", "transgrp/data/trans40au.grp", "transgrp/data/trans42u.grp", "transgrp/data/trans36l.grp",
  "transgrp/data/trans40j.grp", "transgrp/data/trans42b.grp", "transgrp/data/trans39a.grp", "transgrp/data/trans36y.grp",
  "transgrp/data/trans35b.grp", "transgrp/data/trans42w.grp", "transgrp/data/trans40h.grp", "transgrp/data/trans36n.grp",
  "transgrp/data/trans42ab.grp", "transgrp/data/trans40bl.grp", "transgrp/data/trans40aw.grp", "transgrp/data/trans13.grp",
  "transgrp/data/trans12.grp", "transgrp/data/trans40aa.grp", "transgrp/data/trans40av.grp", "transgrp/data/trans42ac.grp",
  "transgrp/data/trans40i.grp", "transgrp/data/trans36o.grp", "transgrp/data/trans42v.grp", "transgrp/data/trans36x.grp",
  "transgrp/data/trans42a.grp", "transgrp/data/trans42r.grp", "transgrp/data/trans36k.grp", "transgrp/data/trans40m.grp",
  "transgrp/data/trans42e.grp", "transgrp/data/trans43a.grp", "transgrp/data/trans40z.grp", "transgrp/data/trans18b.grp",
  "transgrp/data/trans40ae.grp", "transgrp/data/trans40bi.grp", "transgrp/data/trans40ar.grp", "transgrp/data/trans16.grp",
  "transgrp/data/trans17.grp", "transgrp/data/trans40as.grp", "transgrp/data/trans40bh.grp", "transgrp/data/trans40ad.grp",
  "transgrp/data/trans42af.grp", "transgrp/data/trans18c.grp", "transgrp/data/trans42d.grp", "transgrp/data/trans36j.grp",
  "transgrp/data/trans40l.grp", "transgrp/data/trans42s.grp", "transgrp/data/trans38a.grp", "transgrp/data/trans40n.grp",
  "transgrp/data/trans36h.grp", "transgrp/data/trans42q.grp", "transgrp/data/trans40y.grp", "transgrp/data/trans42f.grp",
  "transgrp/data/trans18a.grp", "transgrp/data/trans42ad.grp", "transgrp/data/trans40af.grp", "transgrp/data/trans40bj.grp",
  "transgrp/data/trans40aq.grp", "transgrp/data/trans29.grp", "transgrp/data/trans15.grp", "transgrp/data/trans14.grp",
  "transgrp/data/trans22a.grp", "transgrp/data/trans28.grp", "transgrp/data/trans40ap.grp", "transgrp/data/trans40bk.grp",
  "transgrp/data/trans40ag.grp", "transgrp/data/trans42ae.grp", "transgrp/data/trans34a.grp", "transgrp/data/trans42g.grp",
  "transgrp/data/trans40x.grp", "transgrp/data/trans42p.grp", "transgrp/data/trans40o.grp", "transgrp/data/trans36i.grp",
  "transgrp/init.g", "transgrp/read.g", "transgrp/README.md", "transgrp/lib/", "transgrp/lib/trans.gd", "transgrp/lib/trans.grp",
  "transgrp/lib/trans.gi", "transgrp/lib/bla", "transgrp/htm/", "transgrp/htm/indxS.htm", "transgrp/htm/chapters.htm",
  "transgrp/htm/indxT.htm", "transgrp/htm/indxA.htm", "transgrp/htm/biblio.htm", "transgrp/htm/indxM.htm", "transgrp/htm/indxN.htm",
  "transgrp/htm/indxO.htm", "transgrp/htm/theindex.htm", "transgrp/htm/CHAP001.htm", "transgrp/htm/indxI.htm",
  "transgrp/doc/manual.ind", "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.apple.quarantine'",
  "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.apple.lastuseddate#PS'", "transgrp/doc/manual.pdf",
  "transgrp/doc/transgrp.tex", "transgrp/doc/manual.tex", "transgrp/doc/manual.six", "transgrp/doc/manual.bbl",
  "transgrp/doc/make_doc", "transgrp/doc/manual.bib", "transgrp/doc/manual.mst", "transgrp/tst/bugfix-2013-05-16-t00291.tst",
  "transgrp/tst/testall.g", "transgrp/tst/trans.tst" ]
~~~

In other words, there are the items:

~~~
"/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.apple.quarantine'",
"/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.apple.lastuseddate#PS'"
~~~

which related to some info in the archive that the version of `tar` on the machine running this, does not recognise.

This is easily fixed by adding:

~~~
topdir := Filtered(topdir, f -> not IsEmpty(f));
~~~

or something similar. This resolves this issue, but I'm not sure if this is the desirable solution or not.

**Comment by hulpke** [comment]
The `transgrp` package is wrapped on a Mac (and no, that cannot be changed), which is likely the source of the xattr files. 
Apparently even setting `COPYFILE_DISABLE=1` does not delete these. So PackageManager needs to be able to cope with them.

**Comment by slel** [comment]
Possibly relevant (from searching the web
for [ tarball xattr quarantine ] and similar):

- https://discussions.apple.com/thread/3145071
- https://apple.stackexchange.com/q/87313
- https://superuser.com/q/61185

From macOS, you could maybe copy the files
to a FAT32 drive, remove `._*` files, and tar there?@

**Comment by hulpke** [comment]
As I said, I already am using COPYFILE_DISABLE. I don't realy want to add contortions such as a different wrapping drive.
Ultimately, this looks to me as a situation `PackageManager` needs to be able to cope with.

**Comment by james-d-mitchell** [comment]
@hulpke I completely agree this is an issue with `PackageManager`, thanks for the comments!

**Comment by james-d-mitchell** [comment]
let me try and see

**Comment by mtorpey** [comment]
Hey @james-d-mitchell!  How did this go in the end?  Is PR #82 a fix?

**Comment by james-d-mitchell** [comment]
Sorry for the extended delay @mtorpey, I can confirm that the PR does fix the issue, thanks!
